src/common/leaky_bucket.cpp

```diff
@@ -79,32 +85,71 @@ double LeakyBucket::Bucket::Add(double tokens) {
     }
 }
 
-void LeakyBucket::Bucket::Leak(uint64_t intervalUs) {
+void LeakyBucket::Bucket::BucketInitial(bool burst) {
+    if (burst) {
+        if (abs(burstLevel) <= 0.1) {
+            burstBucketEmptyTimes++;
+            if (burstBucketEmptyTimes >= FLAGS_bucketEmptyTimes) {
+                burstLevelInitial = true;
+                burstLevel = burst - static_cast<double>(burst) *
+                  FLAGS_bucketLeakIntervalMs /
+                    TimeUtility::MilliSecondsPerSecond;
+                burstBucketEmptyTimes = 0;
+            }
+        }
+    } else {
+        if (abs(level) <= 0.1) {
+            avgBucketEmptyTimes++;
+            if (avgBucketEmptyTimes >= FLAGS_bucketEmptyTimes) {
+                    levelInitial = true;
+                    level = avg - static_cast<double>(avg) *
+                      FLAGS_bucketLeakIntervalMs
+                      / TimeUtility::MilliSecondsPerSecond;
+                    avgBucketEmptyTimes = 0;
+            }
+        }
+    }
+}
+
+void LeakyBucket::Bucket::Leak() {
     if (avg == 0) {
         level = 0;
         burstLevel = 0;
         return;
     }
 
-    double leak = static_cast<double>(avg) * intervalUs /
-                  TimeUtility::MicroSecondsPerSecond;
-    level = std::max(level - leak, 0.0);
+    BucketInitial();
+    if (levelInitial)
+        return;
 
+    double leak = static_cast<double>(avg) * FLAGS_bucketLeakIntervalMs /
+                  TimeUtility::MilliSecondsPerSecond;
+    level = std::max(level - leak, 0.0);
+    VLOG(9) << "leak is: " << leak
+            << ", level is: " << level;
     if (burst > 0) {
-        leak = static_cast<double>(burst) * intervalUs /
-               TimeUtility::MicroSecondsPerSecond;
+        BucketInitial(true);
+        if (burstLevelInitial)
+            return;
+
+        leak = static_cast<double>(burst) * FLAGS_bucketLeakIntervalMs /
+               TimeUtility::MilliSecondsPerSecond;
         burstLevel = std::max(burstLevel - leak, 0.0);
+        VLOG(9) << "leak is: " << leak
+                << ", burstLevel is: " << burstLevel;
     }
 }
 
 void LeakyBucket::Bucket::Reset(uint64_t avg, uint64_t burst,
                                 uint64_t burstLength) {
-    level = 0;
-    burstLevel = 0;
-
+    level = avg - static_cast<double>(avg) * FLAGS_bucketLeakIntervalMs /
+            TimeUtility::MilliSecondsPerSecond;
+    burstLevel = burst - static_cast<double>(burst) * FLAGS_bucketLeakIntervalMs
+      / TimeUtility::MilliSecondsPerSecond;
     this->avg = avg;
     this->burst = burst;
     this->burstLength = burstLength;
+    this->capacity = burst + (burst - avg) * (burstLength - 1);
```

Comment:
if so, `burstLength` should rename to `burstSeconds`

Reply:
fix